### PR TITLE
Replace gym with gymnasium

### PR DESCRIPTION
**Pull request by Adaickalavan**
1. This pull request is a step towards replacing gym with gymnasium. Related issue #1981.
   - `info` returned by `hiway-v1` in `reset()` and `step()` methods are unified.
   - Seed of `hiway-v1` env can be retrieved through a new property `seed`.
   - Replaced instances of `hiway-v0` and `gym` with `hiway-v1` and `gymnasium`, respectively.

1. The following are the remaining components which still use gym. They are to be removed or made optional in subsequent pull requests.
    - SMARTS/smarts/env/hiway_env.py
    - SMARTS/smarts/env/tests/test_hiway_env.py
    - SMARTS/smarts/env/gymnasium/wrappers/api_reversion.py
    - SMARTS/examples/rl/rllib
    - SMARTS/smarts/env/wrappers/gif_recorder.py
    - SMARTS/smarts/env/wrappers/record_video.py
    - SMARTS/smarts/env/wrappers/recorder_wrapper.py
    - SMARTS/zoo/evaluation/npc_ego_example.py
    - SMARTS/zoo/policies/cross-rl-agent
    - SMARTS/zoo/policies/rl-agent

### CODE REVIEW

File: smarts/env/gymnasium/hiway_env_v1.py [hunk 1]
```diff
@@ -494,3 +504,12 @@ def smarts(self):
...
         return self._smarts
+
+    @property
+    def seed(self):
+        """Returns the environment seed.
+
+        Returns:
+            int: Environment seed.
+        """
+        return current_seed()
```
Comment (Gamenot) [inline smarts/env/gymnasium/hiway_env_v1.py:515]: I am not sure if the fact that the seed is global for the current thread should be elevated to an issue.

So long as it ensures determinism for the space of a run it is currently enough.

File: smarts/env/tests/test_social_agent.py [hunk 2]
```diff
@@ -19,14 +19,16 @@
...
 import pytest
 
 from smarts.core.agent_interface import AgentInterface, AgentType
-from smarts.env.hiway_env import HiWayEnv
+from smarts.env.gymnasium.hiway_env_v1 import HiWayEnvV1
+from smarts.env.utils.action_conversion import ActionOptions
+from smarts.env.utils.observation_conversion import ObservationOptions
 
 AGENT_ID = "Agent-007"
-SOCIAL_AGENT_ID = "Alec Trevelyan"
```
Comment (Gamenot) [inline smarts/env/tests/test_social_agent.py:29]: I see you are getting rid of the traitor. 🤣

**Review by Gamenot — APPROVED** [review]
I am seeing that this is a step rather than a full replacement (it does not remove `gym` entirely.)

File: smarts/env/tests/test_social_agent.py (hunk 2 above)
Comment (Adaickalavan) [reply smarts/env/tests/test_social_agent.py:29]: Because the traitor was not being tested properly ...

**Pull request merged**